src/components/Button/Button.tsx

```diff
@@ -1,13 +1,80 @@
-import { Text, TouchableOpacity } from 'react-native'
+import React from 'react'
+import { MaterialIcons } from '@expo/vector-icons'
+import { View } from 'react-native'
+import {
+  Button as PaperButton,
+  ButtonProps as PaperButtonProps,
+} from 'react-native-paper'
 
-import { ButtonProps } from './Button.types'
+export interface ButtonProps
+  extends Omit<PaperButtonProps, 'icon' | 'children'> {
+  title?: string | React.ReactNode
+  onPress?: () => void
+  materialIconName?: keyof typeof MaterialIcons.glyphMap
+  icon?: React.ReactNode
+  alignIcon?: 'left' | 'right'
+  loading?: boolean
+}
+
+export default function Button({
+  title,
+  icon,
+  materialIconName,
+  alignIcon = 'left',
+  loading = false,
+  onPress,
+  mode = 'contained',
+  ...rest
+}: ButtonProps) {
+  const handleIcon = React.useCallback(() => {
+    if (icon) {
+      return icon
+    }
+    if (materialIconName) {
+      return <MaterialIcons name={materialIconName} size={20} color={'#fff'} />
+    }
+    return undefined
+  }, [icon, materialIconName])
+
+  const handleIconGap = React.useCallback((): number => {
+    const icon = handleIcon()
+    if (icon && title) {
+      return 12
+    }
+    if (title) {
+      return 0
+    }
+    if (icon) {
+      return -12
+    }
+    return 0
```

Comment:
Queremos esse úlitmo caso em algum momento? Ou seja, um botão sem `icon` e sem `title`? Não conheço muito do projeto, mas não me parece algo que deveria acontecer. Quem sabe até podemos ajustar os tipos pra não ter esse caso. De qualquer jeito, podemos tirar um `if`
```suggestion
  const handleIconGap = React.useCallback((): number => {
    const icon = handleIcon()
    if (icon && title) {
      return 12
    }
    if (icon) {
      return -12
    }
    return 0
```

Reply:
Mesma sugestão de usar ternário ou (de preferência) ts-pattern aqui

Reply:
a princípio não queremos kkk foi na emoção, vou ajustar isso